src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/AbstractTablesTab.java

```diff
@@ -306,14 +326,87 @@ public void invoke(String wikiPageId) {
 				}
 			};
 			wikiPageWidget.setWikiReloadHandler(wikiReloadHandler);
-
+			getLatestSnapshotVersionNumber()
+					.addCallback(new FutureCallback<Long>() {
+						@Override
+						public void onSuccess(@NullableDecl Long result) {
+							latestSnapshotVersionNumber = result;
+							configureVersionAlert();
+						}
+
+						@Override
+						public void onFailure(Throwable t) {
+							synAlert.showError(t.getMessage());
+						}
+					},
+					directExecutor());
 		} else if (isProject) {
 			areaToken = null;
 			tableListWidget.configure(bundle, getTypesShownInList());
 			view.setWikiPageVisible(false);
 			showProjectLevelUI();
+			configureVersionAlert();
 		}
 	}
 
+	private void configureVersionAlert() {
+		boolean versionHistoryIsVisible = this.metadata.getVersionHistoryWidget().isVisible();
+		this.view.setVersionAlertPrimaryAction((versionHistoryIsVisible ? "Hide" : "Show") + " Version History",
+				e -> {
+					this.metadata.getVersionHistoryWidget().setVisible(!versionHistoryIsVisible);
+					// Reconfigure the version alert to update the button text.
+					configureVersionAlert();
+				});
+		if (entityBundle.getEntity() instanceof Dataset && ((Dataset) entityBundle.getEntity()).getIsLatestVersion()) {
+			// This is the 'draft' version of the dataset
+			this.view.setVersionAlertVisible(true);
+			this.view.setVersionAlertCopy(VERSION_ALERT_DRAFT_DATASET_TITLE, VERSION_ALERT_DRAFT_DATASET_MESSAGE);
+			if (latestSnapshotVersionNumber == null) {
+				// No stable version exists
+				this.view.setVersionAlertSecondaryAction(GO_TO_LATEST_STABLE_VERSION,
+						e -> ginInjector.getGlobalApplicationState()
+								.getPlaceChanger()
+								.goTo(new Synapse(entityBundle.getEntity().getId(),
+										latestSnapshotVersionNumber, EntityArea.DATASETS, null)
+								),
+						false,
+						NO_STABLE_VERSIONS_OF_THIS_DATASET);
+
+			} else {
+				// Notify that a stable version exists and link to it
+				this.view.setVersionAlertSecondaryAction(GO_TO_LATEST_STABLE_VERSION,
```

Comment:
Looks like a copy of code above.  Prefer making differences explicit.
Expecting values like
boolean isStableVersionExists, isSecondaryActionEnabled
And String secondaryActionTooltip